### DISCUSSION TITLE
Approach when there are lots of events?

### DISCUSSION THREAD

**Discussion by tomb1n0** [spatie/icalendar-generator#125] [Q&A]
Thanks for the great package, very intuitive and easy to use!

Is there a recommended approach when dealing with thousands of events? The problem i'm running into at the minute is that a single Calendar object is built up and thousands of events added to it, running out of memory.

For example:

```
        $calendar = Calendar::create('Care Coordinator')->refreshInterval(5);

        // Event model has a "toIcalEvent" method on it, here we've got the potential to run out of memory building up this array
        $events = \App\Models\Event::get()->map->toICalEvent()->all();

        $calendar->event($events);
        
        return response($calendar->get())->header('Content-Type', 'text/calendar; charset=utf-8');
```

Would be nice if we could chunk over a group of events, streaming the response output in some way?

**Comment by rubenvanassche** [comment]
At the moment there isn't one, I also have no idea how to implement this since this package is framework agnostic. If someone finds a good and easy solution for this, I would certainly accept the PR!
